nss/src/passwd/mod.rs

```diff
@@ -0,0 +1,110 @@
+use crate::error;
+use libc::uid_t;
+use libnss::interop::Response;
+use libnss::passwd::{Passwd, PasswdHooks};
+use tonic::Request;
+
+use crate::client::{self, authd};
+use authd::PasswdEntry;
+
+pub struct AuthdPasswd;
+impl PasswdHooks for AuthdPasswd {
+    /// get_all_entries returns all passwd entries.
+    fn get_all_entries() -> Response<Vec<Passwd>> {
+        get_all_entries()
+    }
+
+    /// get_entry_by_uid returns the passwd entry for the given uid.
+    fn get_entry_by_uid(uid: uid_t) -> Response<Passwd> {
+        get_entry_by_uid(uid)
+    }
+
+    /// get_entry_by_name returns the passwd entry for the given name.
+    fn get_entry_by_name(name: String) -> Response<Passwd> {
+        get_entry_by_name(name)
+    }
+}
+
+/// get_all_entries connects to the grpc server and asks for all passwd entries.
+pub fn get_all_entries() -> Response<Vec<Passwd>> {
+    super::RT.block_on(async {
+        let mut client = match client::new_client().await {
+            Ok(c) => c,
+            Err(e) => {
+                error!("could not connect to gRPC server: {}", e);
+                return Response::Unavail;
+            }
+        };
+
+        let req = Request::new(authd::Empty {});
+        match client.get_passwd_entries(req).await {
+            Ok(r) => Response::Success(passwd_entries_to_passwds(r.into_inner().entries)),
+            Err(e) => {
+                error!("error when listing passwd: {}", e);
+                Response::NotFound
+            }
+        }
+    })
+}
+
+/// get_entry_by_uid connects to the grpc server and asks for the passwd entry with the given uid.
+pub fn get_entry_by_uid(uid: uid_t) -> Response<Passwd> {
+    super::RT.block_on(async {
+        let mut client = match client::new_client().await {
+            Ok(c) => c,
+            Err(e) => {
+                error!("could not connect to gRPC server: {}", e);
+                return Response::Unavail;
+            }
+        };
+
+        let req = Request::new(authd::GetByIdRequest { id: uid });
+        match client.get_passwd_by_uid(req).await {
+            Ok(r) => Response::Success(passwd_entry_to_passwd(r.into_inner())),
+            Err(e) => {
+                error!("error when getting passwd by uid: {}", e);
+                Response::NotFound
+            }
+        }
+    })
+}
+
+/// get_entry_by_name connects to the grpc server and asks for the passwd entry with the given name.
+pub fn get_entry_by_name(name: String) -> Response<Passwd> {
+    super::RT.block_on(async {
+        let mut client = match client::new_client().await {
+            Ok(c) => c,
+            Err(e) => {
+                error!("could not connect to gRPC server: {}", e);
+                return Response::Unavail;
```

Comment:
did you double check the response time and what we did in aad-auth?

Reply:
What do you mean by "response time"?

Yes, I checked what we did in `aad-auth`. We only return `Unavail` or  `NotFound`. I can try parsing the error message to look for what we return by the cache when no user is found, wdyt?

Reply:
Sorry, I meant response type.

Don’t parse the error message, we are passing a go pre-defined user value and we should not rely on the string itself. Can we maybe use to GRPC error type with predefined status code?